balloon-hash/src/balloon.rs

```diff
@@ -34,11 +37,15 @@ where
 {
     #[cfg(not(feature = "parallel"))]
     let output = {
-        let mut output = GenericArray::<_, D::OutputSize>::default();
-
-        for thread in 1..=u64::from(params.p_cost.get()) {
-            let hash = hash_internal::<D>(pwd, salt, secret, params, memory_blocks, Some(thread))?;
-            output = output.into_iter().zip(hash).map(|(a, b)| a ^ b).collect();
+        let mut output = hash_internal::<D>(pwd, salt, secret, params, memory_blocks, Some(1))?;
+
+        for thread in 2..=u64::from(params.p_cost.get()) {
+            #[cfg_attr(not(feature = "zeroize"), allow(unused_mut))]
+            let mut hash =
+                hash_internal::<D>(pwd, salt, secret, params, memory_blocks, Some(thread))?;
```

Comment:
You can move `let mut hash` outside of the loop and write something like this:
```rust
let mut hash = Default::default();
for thread in 2..=u64::from(params.p_cost.get()) {
    hash = ...;
    // ...
}
hash.zeroize();
```

Reply:
Done.